sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosException.java

```diff
@@ -258,7 +260,7 @@ CosmosException setDiagnostics(CosmosDiagnostics cosmosDiagnostics) {
 
     @Override
     public String toString() {
-        return getClass().getSimpleName() + "{" + "error=" + cosmosError + ", resourceAddress='"
+        return getClass().getSimpleName() + "{" + "sdkVersion=" + SDK_VERSION + ", error=" + cosmosError + ", resourceAddress='"
```

Comment:
I think this need to go into troubleshooting string, rather than just Exception.toSTring()

@simplynaveen20

Reply:
My initial thought on that is, we don't want our customers to rely on getting diagnostics string and then logging the exception. 
It would be much better to have it in all scenarios. 
For example, below is a very common scenario is : 

```
try {
  some operation. 
} catch (Exception e) {
  logger.error("Exception occurred", e);
}
```

That being said, we can also add this to diagnostics string.

Reply:
diagnostics string should have all the needed info for troubleshooting (including sdk version). However it is good to have it here too. We can consider adding it to troubleshooting string later.

Reply:
Valid point, I can add it to the diagnostics string as well in this PR. Let me see what needs to be done for that.

Reply:
@moderakh @simplynaveen20  - would it make sense to include the complete `user agent string` on diagnostics or just the SDK_VERSION ?

Reply:
In the V3 .NET SDK the exception.ToString() contains the message, stacktrace, and diagnostics.  The CosmosException also has properties to access diagnostics, stacktrace, and message separately.